blockchain/blockchain.go

```diff
@@ -564,94 +550,89 @@ func (b *Blockchain) WriteBlocks(blocks []*types.Block) error {
 	}
 
 	// Validate the chain
-	for i, block := range blocks {
-		// Check the parent numbers
-		if block.Number()-1 != parent.Number {
-			return fmt.Errorf(
-				"number sequence not correct at %d, %d and %d",
-				i,
-				block.Number(),
-				parent.Number,
-			)
-		}
-
-		// Check the parent hash
-		if block.ParentHash() != parent.Hash {
-			return fmt.Errorf("parent hash not correct")
-		}
+	// Check the parent numbers
+	if block.Number()-1 != parent.Number {
+		return fmt.Errorf(
+			"number sequence not correct at %d and %d",
+			block.Number(),
+			parent.Number,
+		)
+	}
 
-		// Verify the header
-		if err := b.consensus.VerifyHeader(parent, block.Header); err != nil {
-			return fmt.Errorf("failed to verify the header: %v", err)
-		}
+	// Check the parent hash
+	if block.ParentHash() != parent.Hash {
+		return fmt.Errorf("parent hash not correct")
+	}
 
-		// Verify body data
-		if hash := buildroot.CalculateUncleRoot(block.Uncles); hash != block.Header.Sha3Uncles {
+	// Verify the header
+	if err := b.consensus.VerifyHeader(parent, block.Header); err != nil {
+		return fmt.Errorf("failed to verify the header: %v", err)
+	}
 
-			return fmt.Errorf(
-				"uncle root hash mismatch: have %s, want %s",
-				hash,
-				block.Header.Sha3Uncles,
-			)
-		}
+	// Verify body data
+	if hash := buildroot.CalculateUncleRoot(block.Uncles); hash != block.Header.Sha3Uncles {
 
-		// TODO, the wrapper around transactions
-		if hash := buildroot.CalculateTransactionsRoot(block.Transactions); hash != block.Header.TxRoot {
+		return fmt.Errorf(
+			"uncle root hash mismatch: have %s, want %s",
+			hash,
+			block.Header.Sha3Uncles,
+		)
+	}
 
-			return fmt.Errorf(
-				"transaction root hash mismatch: have %s, want %s",
-				hash,
-				block.Header.TxRoot,
-			)
-		}
+	// TODO, the wrapper around transactions
```

Comment:
i am not sure what this TODO means, do you know?

Reply:
Honestly I have no idea, this comment was made ~3 years ago.

I've removed it since I can't see the relevance any more:

39e4783